### PR TITLE
 like feature for posts

### PR DESCRIPTION
**Pull request by CloCkWeRX**
Was #696 

Screenshots:
like_ui
like_ui2

### CODE REVIEW

**Comment by CloCkWeRX** [comment]
So this is probably OK to merge as is I think, as it just 'works'. I'm tinkering with the AJAX implementation, but probably have to shift the entire like control to be JS centric (rather than half rails template plus 'remote' calls) - otherwise the logic starts getting duplicated.

**Comment by maco** [comment]
don't mind me, just tripping travis

**Comment by CloCkWeRX** [comment]
Darn it, I think I missed committing the ajax work and its lost to the depths of time :/

**Comment by CloCkWeRX** [comment]
This... is probably OK to go, just constantly stuck in codeclimate.

**Comment by cesy** [comment]
Looks like we never actually got round to reviewing this. Let's do that soon, as this will be a nice one to get live.

**Comment by Br3nda** [comment]
This has the "WIP" title so i've not reviewed. Is it ready?

**Comment by CloCkWeRX** [comment]
The title is a hangover from the last PR mostly.

File: app/controllers/likes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
+class LikesController < ApplicationController
+  before_action :authenticate_member!, except: :index
+
+  respond_to :html, :json
+
+  def create
```
Comment (Br3nda) [inline app/controllers/likes_controller.rb]: this method is rather long. Could it be split up?

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      else
+        format.html do
+          flash[:error] = 'Unable to like'
+          redirect_to @like.likeable
+        end
+      end
+    end
+  end
+
+  def destroy
```
Comment (Br3nda) [inline app/controllers/likes_controller.rb]: same again. Can be smaller? Some of the json rendering can go elsewhere and therefore be standardised.

File: app/models/post.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 class Post < ActiveRecord::Base
   extend FriendlyId
+  include Likeable
```
Comment (Br3nda) [inline app/models/post.rb]: :sparkles:  This is super choice as.

File: app/views/posts/_single.html.haml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 .well
-  .post
+  .post{id: "post-#{post.id}"}
```
Comment (Br3nda) [inline app/views/posts/_single.html.haml]: There's a PR attempting to get haml-lint into our CI soon. Can you run a haml-linter over it?

```
app/views/posts/_single.html.haml:2 [W] SpaceInsideHashAttributes: Hash attribute should start with one space after the opening brace
app/views/posts/_single.html.haml:2 [W] SpaceInsideHashAttributes: Hash attribute should end with one space before the closing brace
app/views/posts/_single.html.haml:5 [W] RuboCop: Use the new Ruby 1.9 hash syntax.
app/views/posts/_single.html.haml:5 [W] RuboCop: Use the new Ruby 1.9 hash syntax.
app/views/posts/_single.html.haml:5 [W] RuboCop: Use the new Ruby 1.9 hash syntax.
app/views/posts/_single.html.haml:5 [W] LineLength: Line is too long. [84/80]
app/views/posts/_single.html.haml:30 [W] RuboCop: Use the new Ruby 1.9 hash syntax.
app/views/posts/_single.html.haml:32 [W] SpaceBeforeScript: The - symbol should have one space separating it from code
app/views/posts/_single.html.haml:33 [W] RuboCop: Use the new Ruby 1.9 hash syntax.
app/views/posts/_single.html.haml:35 [W] SpaceBeforeScript: The - symbol should have one space separating it from code
app/views/posts/_single.html.haml:42 [W] LineLength: Line is too long. [132/80]
app/views/posts/_single.html.haml:46 [W] LineLength: Line is too long. [115/80]
```

File: db/schema.rb [hunk 5]
```diff
@@ -433,18 +452,26 @@
...
-    t.datetime "created_at"
-    t.datetime "updated_at"
+    t.datetime "created_at",  null: false
+    t.datetime "updated_at",  null: false
     t.string   "slug"
   end
 
   add_index "roles", ["slug"], name: "index_roles_on_slug", unique: true, using: :btree
 
+<<<<<<< HEAD
```
Comment (Br3nda) [inline db/schema.rb]: Merge conflict~~ :warning:

File: db/schema.rb [hunk 6]
```diff
@@ -433,18 +452,26 @@
...
+  create_table "scientific_names", force: true do |t|
+    t.string   "scientific_name", null: false
+    t.integer  "crop_id",         null: false
+    t.datetime "created_at",      null: false
+    t.datetime "updated_at",      null: false
+=======
```
Comment (Br3nda) [inline db/schema.rb]: Merge conflict~~ :warning:

File: db/schema.rb [hunk 7]
```diff
@@ -433,18 +452,26 @@
...
   create_table "scientific_names", force: :cascade do |t|
     t.string   "name",       null: false
     t.integer  "crop_id",    null: false
     t.datetime "created_at"
     t.datetime "updated_at"
+>>>>>>> upstream/dev
```
Comment (Br3nda) [inline db/schema.rb]: Merge conflict~~ :warning:

File: db/schema.rb [hunk 8]
```diff
@@ -20,16 +20,16 @@
     t.string   "name",              null: false
```
Comment (Br3nda) [inline db/schema.rb]: Can you regenerate this whole file just to make sure?

Comment (Br3nda) [inline db/schema.rb]: After running db:migrate, i get lots of checks on schema.db.

File: db/schema.rb (hunk 5 above)
Comment (pozorvlak) [inline db/schema.rb]: There's another one further up, too. I see we run `rake db:migrate` before `rspec` in Travis, so `schema.rb` gets ignored. We should probably change that, somehow - run rspec once with the existing `schema.rb` and then check that `rake db:migrate` doesn't change it? I don't think the changes it makes to `schema.rb` are guaranteed to be idempotent, unfortunately :-(

File: db/schema.rb [hunk 9]
```diff
@@ -327,22 +340,28 @@
...
     t.integer  "product_id"
     t.integer  "price"
     t.integer  "quantity"
-    t.datetime "created_at"
-    t.datetime "updated_at"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
   end
 
+<<<<<<< HEAD
```
Comment (pozorvlak) [inline db/schema.rb]: Merge conflict.

File: app/controllers/likes_controller.rb (hunk 2 above)
Comment (pozorvlak) [inline app/controllers/likes_controller.rb]: I was going to say that L16-L21 can go in `Like.to_json` and be re-used, but I guess that's really view code so should go somewhere under `app/views`. What's best practice here?

File: spec/support/is_likeable.rb [hunk 10]
```diff
@@ -0,0 +1,24 @@
+shared_examples "it is likeable" do
+  before(:each) do
+    # Possibly a horrible hack.
+    # Will fail if factory name does not match the model name..
```
Comment (pozorvlak) [inline spec/support/is_likeable.rb]: Super-tiny nitpick: s/../.

File: spec/support/is_likeable.rb [hunk 11]
```diff
@@ -0,0 +1,24 @@
+shared_examples "it is likeable" do
+  before(:each) do
+    # Possibly a horrible hack.
```
Comment (pozorvlak) [inline spec/support/is_likeable.rb]: Nah, seems pretty reasonable to me. I think the default factory for all our models is the same as the model name - if not, it probably should be!

**Comment by Br3nda** [comment]
I've send a PR to this PR.

**Pull request closed without merging**